cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -313,37 +313,49 @@ impl LockedManifestCatalog {
         seed_lockfile: Option<&LockedManifestCatalog>,
         client: &impl catalog::ClientTrait,
     ) -> Result<LockedManifestCatalog, LockedManifestError> {
-        let groups = Self::collect_package_groups(manifest, seed_lockfile).collect();
-
-        // lock existing packages
+        let groups = Self::collect_package_groups(manifest, seed_lockfile);
+        let (already_locked_packages, groups_to_lock) =
+            Self::split_fully_locked_groups(groups, seed_lockfile);
+
+        if groups_to_lock.is_empty() {
+            debug!("All packages are already locked, skipping resolution");
+            return Ok(LockedManifestCatalog {
+                version: Version::<1>,
+                manifest: manifest.clone(),
+                packages: already_locked_packages,
+            });
+        }
 
+        // lock packages
         let resolved = client
-            .resolve(groups)
+            .resolve(groups_to_lock)
             .await
             .map_err(LockedManifestError::CatalogResolve)?;
 
+        // unpack locked packages from response
         let locked_packages = Self::locked_packages_from_resolution(manifest, resolved)?.collect();
 
         let lockfile = LockedManifestCatalog {
             version: Version::<1>,
             manifest: manifest.clone(),
-            packages: locked_packages,
+            packages: [already_locked_packages, locked_packages].concat(),
         };
 
         Ok(lockfile)
     }
 
-    /// Transform a lockfile into a mapping  that is easier to query:
-    /// Lockfile -> { (package, system): locked package }
+    /// Transform a lockfile into a mapping that is easier to query:
+    /// Lockfile -> { (install_id, system): (package_descriptor, locked_package) }
     fn make_seed_mapping(
         seed: &LockedManifestCatalog,
-    ) -> HashMap<(&ManifestPackageDescriptor, &System), &LockedPackageCatalog> {
+    ) -> HashMap<(&String, &System), (&ManifestPackageDescriptor, &LockedPackageCatalog)> {
```

Comment:
Does it make sense for us to make an `InstallId` newtype? It's not immediately obvious from the signature what this `String` is supposed to be.

Reply:
Yes i think it would make sense but i think it would cause a bit much noise changing that here.